src/eva/vision/data/datasets/classification/camelyon16.py

```diff
@@ -0,0 +1,228 @@
+            if not os.path.isdir(os.path.join(self._root, resource)):
+                raise FileNotFoundError(f"'{resource}' not found in the root folder.")
+
+        if not os.path.isfile(os.path.join(self._root, "testing/reference.csv")):
+            raise FileNotFoundError("'reference.csv' file not found in the testing folder.")
+
+    @override
+    def validate(self) -> None:
+
+        expected_n_files = {
+            "train": 216,
+            "val": 54,
+            "test": 129,
+            None: 399,
+        }
+        length = expected_n_files.get(self._split, 0)
+        _validators.check_number_of_files(self._file_paths, length, self._split)
+
+        _validators.check_dataset_integrity(
+            self,
+            length=None,
+            n_classes=2,
+            first_and_last_labels=("normal", "tumor"),
+        )
+
+    @override
+    def filename(self, index: int) -> str:
+        return os.path.basename(self._file_paths[self._get_dataset_idx(index)])
+
+    @override
+    def __getitem__(self, index: int) -> Tuple[np.ndarray, np.ndarray, Dict[str, Any]]:
+        return base.ImageClassification.__getitem__(self, index)
+
+    @override
+    def load_image(self, index: int) -> torch.Tensor:
+        return wsi.MultiWsiDataset.__getitem__(self, index)
+
+    @override
+    def load_target(self, index: int) -> np.ndarray:
+        file_path = self._file_paths[self._get_dataset_idx(index)]
+        return np.asarray(self._get_target_from_path(file_path))
+
+    @override
+    def load_metadata(self, index: int) -> Dict[str, Any]:
+        return {"wsi_id": self.filename(index).split(".")[0]}
+
+    def _load_file_paths(self, split: Literal["train", "val", "test"] | None = None) -> List[str]:
+        """Loads the file paths of the corresponding dataset split."""
+        train_paths, val_paths = [], []
+        for path in glob.glob(os.path.join(self._root, "training/**/*.tif")):
+            if self._get_id_from_path(path) in self._val_slides:
+                val_paths.append(path)
+            else:
+                train_paths.append(path)
+        test_paths = sorted(glob.glob(os.path.join(self._root, "testing/images", "*.tif")))
+
+        match split:
+            case "train":
+                paths = train_paths
+            case "val":
+                paths = val_paths
+            case "test":
+                paths = test_paths
+            case None:
+                paths = train_paths + val_paths + test_paths
+            case _:
+                raise ValueError("Invalid split. Use 'train', 'val' or `None`.")
+        return [os.path.relpath(path, self._root) for path in paths]
+
+    def _get_target_from_path(self, file_path: str) -> int:
+        """Returns the target label based on the file path."""
+        slide_id = self._get_id_from_path(file_path)
+        test_annotations = self.annotations_test_set.get(slide_id, "")
+
+        for class_name in self.classes:
```

Comment:
Would be good if we could get rid of the for loop here, the proposal [here](https://github.com/kaiko-ai/eva/pull/476#discussion_r1626074067) would be one option to do so.